api/app/Models/PoolCandidate.php

```diff
@@ -675,8 +675,12 @@ public function isDraft()
      */
     public function scopeAuthorizedToView(Builder $query)
     {
-        $userId = Auth::user()->id;
-        $user = User::find($userId);
+        $authUser = Auth::user();
+        if (! $authUser) {
+            return $query->where('id', null);
+        }
+
+        $user = User::find($authUser->id);
```

Comment:
Aren't $authUser and $user identical? What's the purpose of this line?

Reply:
`Auth::user()` was giving the error `Undefined method 'isAbleTo'`. Though I do see we have that in `User.php` and just ignore it. I assume it is safe to ignore that error? 🤔

Reply:
Technically, what we get back is an Authenticatable, not a User.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/550a6e33-6f0d-411d-84b8-adaa7e8fd2e3)

In practice, for us, that's always a user but maybe it won't have the Eloquent relationships there?

Reply:
Darn, I see! Does php actually throw an error, or is it just linting which _thinks_ it will?
My main concern is if this forces a repeat of the database call.
Looking around online, I haven't found any suggestions for how to change the return type of Auth::user. Some people have suggested that `$user = User::find(Auth::id())` _might_ be more efficient, if we happen to have the id in memory but not the user yet, but I doubt that's the case for us.

Reply:
So, we _know_ we return a user from our provider. 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c97149e361df11313767f153f06c0535b01b1112/api/config/auth.php#L66-L70

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c97149e361df11313767f153f06c0535b01b1112/api/app/Providers/AuthServiceProvider.php#L98-L117

Could we just type hint this to make the errors go away? I think Intelephense is just flagging this because it is not aware of the middleware which could return anything that extends `Authenticatable` 

```php
/** @var \App\Models\User */
$user = Auth::user();
```

Reply:
Yeah, if it doesn't throw an error at runtime, than I'm fine with typehinting it or ignoring the lint error, etc

Reply:
Cool! Updated in [use type hint to resolve error, prevent additional db hit](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9187/commits/9452cdaf5f77bf630c3cf7ab924ee3ec038a66fe)